sdk/cognitivelanguage/azure-ai-language-questionanswering/swagger/README.md

```diff
@@ -58,3 +58,65 @@ directive:
         $["operationId"] = "queryText";
 ```
 
+### Rename `KnowledgeBasedQueryOptions` -> `Options`
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $["parameters"]["KnowledgeBaseQueryOptions"]
+    transform: >
+        $["x-ms-client-name"] = "Options";
+```
+
+### Rename `TextQueryOptions` -> `Options`
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $["parameters"]["TextQueryOptions"]
+    transform: >
+        $["x-ms-client-name"] = "Options";
+```
+
+### Rename `KnowledgeBaseQueryOptions` -> `QueryKnowledgeBaseOptions`
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $["definitions"]["KnowledgeBaseQueryOptions"]
+    transform: >
+        $["x-ms-client-name"] = "QueryKnowledgeBaseOptions";
+```
+
+### Rename `TextQueryOptions` -> `QueryTextOptions`
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $["definitions"]["TextQueryOptions"]
+    transform: >
+        $["x-ms-client-name"] = "QueryTextOptions";
+```
+
+### Delete `StringIndexType`
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $["definitions"]["TextQueryOptions"]
+    transform: >
+        delete $.properties["stringIndexType"]
+```
+
+### Make `MetadataFilter`'s `metadata` property a list of string
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $["definitions"]
+    transform: >
+        delete $["MetadataFilter"]["properties"]["metadata"]["items"]["$ref"];
```

Comment:
we actually want a list of tuple of string, but autorest can't handle tuples yet, so we're going to transform it into a list of list of string for easier code changes post-generation